docs/standard/serialization/system-text-json/source-generation-modes.md

```diff
@@ -94,34 +94,68 @@ In both reflection and source generation modes:
 
 For information about other known issues with source generation, see the [GitHub issues that are labeled "source-generator"](https://github.com/dotnet/runtime/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-System.Text.Json+label%3Asource-generator) in the *dotnet/runtime* repository.
 
 ## Serialization optimization mode
 
-`JsonSerializer` has many features that customize the output of serialization, such as [camel-casing property names](customize-properties.md#use-camel-case-for-all-json-property-names) and [preserving references](preserve-references.md#preserve-references-and-handle-circular-references). Support for all those features causes some performance overhead. Source generation can improve serialization performance by generating optimized code that uses [`Utf8JsonWriter`](use-utf8jsonwriter.md) directly.
+`JsonSerializer` has features that customize the output of serialization, such as [camel-casing property names](customize-properties.md#use-camel-case-for-all-json-property-names) and [preserving references](preserve-references.md#preserve-references-and-handle-circular-references). However, support for these features adds a performance overhead. Source generation can improve serialization performance by generating optimized code that uses [`Utf8JsonWriter`](use-utf8jsonwriter.md) directly.
 
-The optimized code doesn't support all of the serialization features that `JsonSerializer` supports. The serializer detects whether the optimized code can be used and falls back to default serialization code if unsupported options are specified. For example, <xref:System.Text.Json.Serialization.JsonNumberHandling.AllowReadingFromString?displayProperty=nameWithType> is not applicable to writing, so specifying this option doesn't cause a fall-back to default code.
+The optimized code doesn't support all of the serialization features that `JsonSerializer` supports, although in .NET 8 and later versions, most features are supported. The serializer detects whether the optimized code can be used and falls back to default serialization code if unsupported options are specified. For example, <xref:System.Text.Json.Serialization.JsonNumberHandling.AllowReadingFromString?displayProperty=nameWithType> is not applicable to writing, so specifying this option doesn't cause a fall-back to default code.
 
 The following table shows which options in `JsonSerializerOptions` are supported by the optimized serialization code:
 
+:::zone pivot="dotnet-7-0,dotnet-6-0"
+
 | Serialization option                                                   | Supported by optimized code |
 |------------------------------------------------------------------------|-----------------------------|
-| <xref:System.Text.Json.JsonSerializerOptions.Converters>               | ❌                          |
+| <xref:System.Text.Json.JsonSerializerOptions.AllowTrailingCommas>      | ❌                         |
```

Comment:
"Optimized mode" in this case should refer to `JsonSourceGenerationMode.Serialization` or fast-path source gen. Even though these configuration facets were added for source gen in general, these only impact metadata mode and the support matrix for fast-path a.k.a. optimized mode hasn't changed at all since .NET 7